src/api/InfluxDB.cpp

```diff
@@ -0,0 +1,238 @@
+	Buffer::iterator it;
+
+	_api.curl = curlSessionProvider ? curlSessionProvider->get_easy_session(_host + channel()->uuid()) : 0; // TODO add option to use parallel sessions. Simply add uuid() to the key.
+	if (!_api.curl) {
+		throw vz::VZException("CURL: cannot create handle.");
+	}
+
+	print(log_debug, "Buffer has %i items", channel()->name(), buf->size());
+
+	buf->lock();
+	int request_body_lines = 0;
+	for (it = buf->begin(); it != buf->end(); it++) {
+		if(request_body_lines >= _max_batch_inserts) {
+			print(log_debug, "reached maximum lines for InfluxDB insertion request.", channel()->name());
+			break;
+		}
+		print(log_finest, "Reading buffer: timestamp %lld value %f", channel()->name(), it->time_ms(), it->value());
+		request_body.append(_measurement_name);
+		request_body.append(",");
+		request_body.append("uuid=");
+		request_body.append(channel()->uuid());
+		request_body.append(" ");
+		request_body.append("value=");
+		request_body.append(std::to_string(it->value()));
+		request_body.append(" ");
+		request_body.append(std::to_string(it->time_ms()));
+		request_body.append("000000"); // needed for correct InfluxDB timestamp
+		request_body.append("\n");  // each measurement on new line
+		it->mark_delete();
+		request_body_lines++;
+	}
+	buf->unlock();
+	print(log_finest, "request body is %s", channel()->name(), request_body.c_str());
+
+	_response->clear_response(); // initialize with empty response
+
+  // if the username option is set, use curl with HTTP basic auth
+	if(!_username.empty()) {
+		curl_easy_setopt(_api.curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+		curl_easy_setopt(_api.curl, CURLOPT_USERNAME, _username.c_str());
+		curl_easy_setopt(_api.curl, CURLOPT_PASSWORD, _password.c_str());
+	}
+	curl_easy_setopt(_api.curl, CURLOPT_URL, _url.c_str());
+	curl_easy_setopt(_api.curl, CURLOPT_VERBOSE, options.verbosity());
+
+	curl_easy_setopt(_api.curl, CURLOPT_DEBUGFUNCTION, &(vz::api::CurlCallback::debug_callback));
+	curl_easy_setopt(_api.curl, CURLOPT_DEBUGDATA, response());
+	// signal-handling in libcurl is NOT thread-safe. so force to deactivated them!
+	curl_easy_setopt(_api.curl, CURLOPT_NOSIGNAL, 1);
+	curl_easy_setopt(_api.curl, CURLOPT_TIMEOUT, _curl_timeout);
+
+  curl_easy_setopt(_api.curl, CURLOPT_POSTFIELDS, request_body.c_str());
+	curl_easy_setopt(_api.curl, CURLOPT_WRITEFUNCTION, &(vz::api::CurlCallback::write_callback));
+	curl_easy_setopt(_api.curl, CURLOPT_WRITEDATA, response());
+
+	curl_code = curl_easy_perform(_api.curl);
+  print(log_finest, "Influxdb curl terminated", channel()->name());
+	curl_easy_getinfo(_api.curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+	if (curl_code == CURLE_OK && http_code >= 200 && http_code < 300) { // everything is ok
+		print(log_debug, "InfluxDB CURL success", channel()->name());
+	  buf->clean();  // delete the stuff we just sent to InfluxDB from the buffer
+	}
+	else {
+		buf->undelete();  // failure to insert, so dont delete the buffer
```

Comment:
you might add some error handling if it can be never send. e.g. prevent growing above a certain size.